mantle/platform/qemu.go

```diff
@@ -1394,19 +1397,38 @@ func (builder *QemuBuilder) Exec() (*QemuInstance, error) {
 	}
 	argv = append(argv, "-smp", fmt.Sprintf("%d", builder.Processors))
 
-	switch builder.Firmware {
-	case "bios":
-		break
-	case "uefi":
-		if err := builder.setupUefi(false); err != nil {
-			return nil, err
-		}
-	case "uefi-secure":
-		if err := builder.setupUefi(true); err != nil {
-			return nil, err
+	switch coreosarch.CurrentRpmArch() {
+	case "x86_64", "aarch64":
+		// Default to UEFI (but not secure boot)
+		if builder.RequireUEFI {
+			builder.Firmware = "uefi"
+		} else if builder.Firmware == "" {
+			builder.Firmware = "uefi"
+		}
+
+		switch builder.Firmware {
+		case "uefi":
+			if err := builder.setupUefi(false); err != nil {
+				return nil, err
+			}
+		case "uefi-secure":
+			if err := builder.setupUefi(true); err != nil {
+				return nil, err
+			}
+		case "bios":
+			{
+				if coreosarch.CurrentRpmArch() != "x86_64" {
+					panic(fmt.Sprintf("Unknown firmware: %s", builder.Firmware))
```

Comment:
`Firmware` comes from a user-provided option (and this function already returns an `error`) so this isn't a programmer error.  Panicking is inappropriate here.